File: java/spring-hibernate/src/main/java/com/sprint/sms/api/domain/Role.java
package com.sprint.sms.api.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

import com.ideahut.common2.annotation.IdhFormatter;
import com.ideahut.shared2.domain.base.LongIdEntryVersionDomain;

@Entity
@Table(name = "t_role")
@IdhFormatter
public class Role extends LongIdEntryVersionDomain {	
	/**
	 * 
	 */
	private static final long serialVersionUID = -4881491628578757068L;
	
	private String name;
	
	private Boolean active;

	@Column(name = "name", length = 150, nullable = false)
	@IdhFormatter
	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
	
	@Column(name = "active", nullable = false)
	@IdhFormatter
	public Boolean getActive() {
		return active;
	}

	public void setActive(Boolean active) {
		this.active = active;
	}
	
}

File: php/slim-doctrine/app/src/com/sprint/sms/api/domain/Role.php
<?php
namespace App\com\sprint\sms\api\domain;

use Doctrine\ORM\Mapping as ORM;
use \App\com\sprint\sms\api\base\domain\BaseEntryVersion;
use \App\com\sprint\sms\api\support\FormatSupport;

/**
 * @ORM\Entity
 * @ORM\Table(name = "t_role")
 */
class Role extends BaseEntryVersion implements FormatSupport
{

	/**
     * @ORM\Column(name = "ID_", type = "bigint")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy = "IDENTITY")
	 */
	private $id;
	
	/**
     * @ORM\Column(name = "name", type = "string", length = 150, nullable = false)
     */
    private $name;
	
	/**
     * @ORM\Column(name = "active", type = "boolean", nullable = false, options = {"default":true})
     */
    private $active;
	
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}   
	
	public function setName($name) {
		$this->name = $name;
	}
	
	public function getName() {
		return $this->name;
	}
	
	public function setActive($active) {
		$this->active = $active;
	}
	
	public function getActive() {
		return $this->active;
	}
	
	
	public function toFormatObject() {
		$o = parent::toFormatObject();
		$o["id"] = $this->id;
		$o["name"] = $this->name;
		$o["active"] = isset($this->active) ? ($this->active ? 1 : 0) : null;	
		return $o;
	}
	
}